lib/spack/spack/directives.py

```diff
@@ -218,6 +218,8 @@ def _execute(pkg):
         if checksum:
             kwargs['md5'] = checksum
 
+        kwargs['version'] = ver
```

Comment:
It's still mysterious to me as well. I believe [this function](https://github.com/LLNL/spack/blob/develop/lib/spack/spack/fetch_strategy.py#L133) determines which FetchStrategy is used, although I'm not sure where it is instantiated.